projects/thales/utils.js

```diff
@@ -0,0 +1,80 @@
+const { sumTokens2 } = require('../helper/unwrapLPs')
+const abi = require('./abi.json')
+const CHAIN_CONFIG = require('./chainConfig')
+
+const utils = {
+  // Get token address from chain config
+  getTokenAddress: (chain, symbol) => CHAIN_CONFIG[chain].tokens[symbol],
+
+  // Add LP TVL
+  addLPTvl: async (api, contract, token) => {
+    const tokenAddress = utils.getTokenAddress(api.chain, token)
+    api.add(tokenAddress, await api.call({
+      target: contract,
+      abi: abi.totalDeposited
+    }))
+  },
+
+  // Get markets from manager
+  getMarketsFromManager: async (api, manager, abiMethod) => {
+    return api.call({
+      target: manager,
+      abi: abi[abiMethod],
+      params: [0, 1000]
+    })
+  },
+
+  // Get all markets for a chain
+  getAllMarkets: async (api, chain) => {
+    const config = CHAIN_CONFIG[chain]
+    if (!config.managers) return []
+
+    const [digitalMarkets, sportsMarkets] = await Promise.all([
+      Promise.all((config.managers.digital || []).map(manager =>
+        utils.getMarketsFromManager(api, manager, 'activeMarkets')
+      )),
+      Promise.all((config.managers.sports || []).map(manager =>
+        utils.getMarketsFromManager(api, manager, 'getActiveTickets')
+      ))
+    ])
+
+    return [...digitalMarkets.flat(), ...sportsMarkets.flat()]
+  },
+
+  // Calculate pool2 TVL
+  getPool2Tvl: async (api, chain) => {
+    const config = CHAIN_CONFIG[chain].pool2
```

Comment:
thanks for the PR but pool2 is, value of LP tokens deposited in your contract (usually masterchefs) not value of tokens on LP, it belongs to dex

Reply:
Thank you for the clarification. I removed the pool2 related logic.